src/zenml/config/global_config.py

```diff
@@ -13,19 +13,244 @@
+            return cast("GlobalConfig", super().__call__(*args, **kwargs))
+
+        if not cls._global_config:
+            cls._global_config = cast(
+                "GlobalConfig", super().__call__(*args, **kwargs)
+            )
+            cls._global_config._migrate_config()
+            cls._global_config._add_and_activate_default_profile()
+
+        return cls._global_config
+
+
+class BaseGlobalConfiguration(ABC):
+    """Base class for global configuration management.
+
+    This class defines the common interface related to profile and stack
+    management that all global configuration classes must implement.
+    Both the GlobalConfig and Repository classes implement this class, since
+    they share similarities concerning the management of active profiles and
+    stacks.
+    """
+
+    @abstractmethod
+    def activate_profile(self, profile_name: str) -> None:
+        """Set the active profile
+
+        Args:
+            profile_name: The name of the profile to set as active.
 
+        Raises:
+            KeyError: If the profile with the given name does not exist.
+        """
+
+    @property
+    @abstractmethod
+    def active_profile(self) -> Optional["ConfigProfile"]:
+        """Return the profile set as active for the repository.
+
+        Returns:
+            The active profile or None, if no active profile is set.
+        """
+
+    @property
+    @abstractmethod
+    def active_profile_name(self) -> Optional[str]:
+        """Return the name of the profile set as active.
+
+        Returns:
+            The active profile name or None, if no active profile is set.
+        """
+
+    @abstractmethod
+    def activate_stack(self, stack_name: str) -> None:
+        """Set the active stack for the active profile.
+
+        Args:
+            stack_name: name of the stack to activate
+
+        Raises:
+            KeyError: If the stack with the given name does not exist.
+        """
+
+    @property
+    @abstractmethod
+    def active_stack_name(self) -> Optional[str]:
+        """Get the active stack name from the active profile.
+
+        Returns:
+            The active stack name or None if no active stack is set or if
+            no active profile is set.
+        """
+
+
+def get_default_store_type() -> StoreType:
+    """Return the default store type.
+
+    The default store type can be set via the environment variable
+    ZENML_DEFAULT_STORE_TYPE. If this variable is not set, the default
+    store type is set to 'LOCAL'.
+
+    NOTE: this is a global function instead of a default
+    `ConfigProfile.store_type` value because it makes it easier to mock in the
+    unit tests.
+
+    Returns:
+        The default store type.
+    """
+    store_type = os.getenv(ENV_ZENML_DEFAULT_STORE_TYPE)
+    if store_type and store_type in StoreType.values():
+        return StoreType(store_type)
+    return StoreType.LOCAL
 
-class GlobalConfig(BaseModel):
+
+class ConfigProfile(BaseModel):
```

Comment:
This is already a huge file, maybe this class could be moved to its own file? (and perhaps renamed, as I suggested below)